### PR TITLE
1.4.2 Release Candidate

### PR DESCRIPTION
**Pull request by Znote**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Cherry-picking bugfixes into 1.4.1 from master to make 1.4.2

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Comment by Znote** [comment]
I want to wait for #3915 to get merged to master, and probably cherry-pick it here. So no need to submit approval reviews here yet.
This list of bugfixes was sent to me by @EPuncker, if there is anything I have missed that should go into 1.4.2, please let me know.

File: data/lib/core/player.lua
Comment (EPuncker) [inline data/lib/core/player.lua:336]: this need to be removed

**Review by EPuncker — CHANGES_REQUESTED** [review]

File: src/luascript.h
Comment (EPuncker) [inline src/luascript.h:27]:
```suggestion
```

**Review by EPuncker — CHANGES_REQUESTED** [review]
just a minor request change and I think its ready

**Review by EPuncker — APPROVED** [review]

**Comment by Codinablack** [comment]
I have been using this branch to develop on for more than a couple months now, for so long that I was confident enough, it was stable enough, for me to build on top of it, my own server. So I am just saying, this is a great candidate for 1.4.2 I had zero problems with it during its use.

**Review by Codinablack — APPROVED** [review]

**Review by DSpeichert — APPROVED** [review]

**Pull request merged**
